stdlib/public/Python/NumpyConversion.swift

```diff
@@ -0,0 +1,180 @@
+//===-- NumpyConversion.swift ---------------------------------*- swift -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// This file defines the `NumpyArrayConvertible` protocol for bridging
+// `numpy.ndarray`.
+//
+//===----------------------------------------------------------------------===//
+
+/// A type that can be initialized from a `numpy.ndarray` instance represented
+// as a `PyValue`.
+public protocol NumpyArrayConvertible {
+  init?(numpyArray: PyValue)
+}
+
+/// A type that is compatible with a NumPy scalar `dtype`.
+public protocol NumpyScalarCompatible {
+  static func isCompatibleNumpyScalarType(_ dtype: PyValue) -> Bool
+}
+
+extension Bool : NumpyScalarCompatible {
+  public static func isCompatibleNumpyScalarType(_ dtype: PyValue) -> Bool {
+    guard let np = try? Python.attemptImport("numpy") else { return false }
```

Comment:
- I don't think `attemptImport` is the best way here. When you pass in a dtype, numpy being existent should be a strict precondition. Just `import` would be fine.
- No matter the python runtime does caching or not, you can make this more efficient by declaring a fileprivate global variable in this file. It's lazily initialized anyway.